Gulp goesn’t use exit code 1 on Mocha errors

I use `gulp-mocha` 0.5.0 have task:

```
gulp.task('test', function () {
    require('./');
    var mocha = require('gulp-mocha');

    return gulp.src('test/*.js', { read: false })
        .pipe(mocha());
});
```

When I start task, I have mocha failed tests, but exit code is 0:

```
postcss → n gulp test
[01:26:51] Using gulpfile ~/Dev/postcss/gulpfile.js
[01:26:51] Starting 'test'...

  [01:26:52] Finished 'test' after 671 ms
․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․

  189 passing (198ms)
  4 failing

  1) source maps changes previous source map:

      AssertionError: expected { source: '/a.css', line: 1, column: 4, name: null } to equal { source: 'a.css', line: 1, column: 4, name: null }
      + expected - actual

       {
         "column": 4,
         "line": 1,
         "name": null,
      +  "source": "a.css"
      -  "source": "/a.css"
       }

      at Assertion.prop.(anonymous function) [as eql] (/home/ai/Dev/postcss/node_modules/should/lib/should.js:61:14)
      at Context.css (/home/ai/Dev/postcss/test/map.js:95:17)
      at callFn (/home/ai/Dev/postcss/node_modules/mocha/lib/runnable.js:223:21)
      at Test.Runnable.run (/home/ai/Dev/postcss/node_modules/mocha/lib/runnable.js:216:7)
      at Runner.runTest (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:373:10)
      at /home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:451:12
      at next (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:298:14)
      at /home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:308:7
      at next (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:246:23)
      at Object._onImmediate (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:275:5)

  2) source maps uses map from subdir:

      AssertionError: expected '/a.css' to equal '../../a.css'
      + expected - actual

      +"../../a.css"
      -"/a.css"

      at Assertion.prop.(anonymous function) [as eql] (/home/ai/Dev/postcss/node_modules/should/lib/should.js:61:14)
      at Context.$__0.doubler.process.from (/home/ai/Dev/postcss/test/map.js:260:24)
      at callFn (/home/ai/Dev/postcss/node_modules/mocha/lib/runnable.js:223:21)
      at Test.Runnable.run (/home/ai/Dev/postcss/node_modules/mocha/lib/runnable.js:216:7)
      at Runner.runTest (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:373:10)
      at /home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:451:12
      at next (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:298:14)
      at /home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:308:7
      at next (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:246:23)
      at Object._onImmediate (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:275:5)

  3) source maps works with different types of maps:

      AssertionError: expected '/a.css' to equal 'a.css'
      + expected - actual

      +"a.css"
      -"/a.css"

      at Assertion.prop.(anonymous function) [as eql] (/home/ai/Dev/postcss/node_modules/should/lib/should.js:61:14)
      at Context.$__0.doubler.process.from (/home/ai/Dev/postcss/test/map.js:319:28)
      at callFn (/home/ai/Dev/postcss/node_modules/mocha/lib/runnable.js:223:21)
      at Test.Runnable.run (/home/ai/Dev/postcss/node_modules/mocha/lib/runnable.js:216:7)
      at Runner.runTest (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:373:10)
      at /home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:451:12
      at next (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:298:14)
      at /home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:308:7
      at next (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:246:23)
      at Object._onImmediate (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:275:5)

  4) source maps miss source content on request:
     Error: "../a.css" is not in the SourceMap.
      at SourceMapConsumer_sourceContentFor [as sourceContentFor] (/home/ai/Dev/postcss/node_modules/source-map/lib/source-map/source-map-consumer.js:375:13)
      at Context.<anonymous> (/home/ai/Dev/postcss/test/map.js:364:29)
      at callFn (/home/ai/Dev/postcss/node_modules/mocha/lib/runnable.js:223:21)
      at Test.Runnable.run (/home/ai/Dev/postcss/node_modules/mocha/lib/runnable.js:216:7)
      at Runner.runTest (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:373:10)
      at /home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:451:12
      at next (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:298:14)
      at /home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:308:7
      at next (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:246:23)
      at Object._onImmediate (/home/ai/Dev/postcss/node_modules/mocha/lib/runner.js:275:5)

postcss →
```

Travis CI also get exit code 0: https://travis-ci.org/ai/postcss/builds/29471330

--- a/index.js
+++ b/index.js
@@ -1,7 +1,7 @@
 'use strict';
 var domain = require('domain');
 var gutil = require('gulp-util');
-var through = require('through2');
+var through = require('through');
 var Mocha = require('mocha');
 
 module.exports = function (options) {
@@ -20,17 +20,16 @@ module.exports = function (options) {
 		}
 	}
 
-	return through.obj(function (file, enc, cb) {
+	return through(function (file) {
 		mocha.addFile(file.path);
-		cb(null, file);
-	}, function (cb) {
+		this.queue(file);
+	}, function () {
 		var stream = this;
 		var d = domain.create();
 
 		function handleException(err) {
 			clearCache();
 			stream.emit('error', new gutil.PluginError('gulp-mocha', err));
-			cb();
 		}
 
 		d.on('error', handleException);
@@ -43,9 +42,9 @@ module.exports = function (options) {
 						stream.emit('error', new gutil.PluginError('gulp-mocha', errCount + ' ' + (errCount === 1 ? 'test' : 'tests') + ' failed.', {
 							showStack: false
 						}));
+					} else {
+						stream.emit('end');
 					}
-
-					cb();
 				});
 			} catch (err) {
 				handleException(err);

--- a/package.json
+++ b/package.json
@@ -35,6 +35,6 @@
   "dependencies": {
     "gulp-util": "^3.0.0",
     "mocha": "^1",
-    "through2": "^0.6.1"
+    "through": "^2.3.4"
   }
 }

--- a/test.js
+++ b/test.js
@@ -1,7 +1,7 @@
+/* global afterEach, it */
 'use strict';
 var assert = require('assert');
 var gutil = require('gulp-util');
-var through = require('through2');
 var mocha = require('./');
 
 var out = process.stdout.write.bind(process.stdout);
@@ -41,29 +41,26 @@ it('should run unit test and fail', function (cb) {
 	stream.end();
 });
 
-it('should call the callback for the flush method', function (cb) {
+it('should call the callback right after end', function (cb) {
 	var stream = mocha();
-
-	stream._flush(function () {
+	stream.on('end', function () {
 		assert(true);
 		cb();
 	});
+	stream.end();
 });
 
 it('should clear cache after successful run', function (done) {
 	var stream = mocha();
 
-	stream.pipe(through.obj(function (file, enc, cb) {
-		cb();
-	}, function (cb) {
+	stream.on('end', function () {
 		for (var key in require.cache) {
 			if (/fixture-pass/.test(key.toString())) {
 				throw new Error('require cache still contained: ' + key);
 			}
 		}
-
 		done();
-	}));
+	});
 
 	stream.write(new gutil.File({path: 'fixture-pass.js'}));
 	stream.end();
@@ -71,21 +68,14 @@ it('should clear cache after successful run', function (done) {
 
 it('should clear cache after failing run', function (done) {
 	var stream = mocha();
-
-	stream.pipe(through.obj(function (file, enc, cb) {
-		cb();
-	}, function (cb) {
+	stream.on('error', function () {
 		for (var key in require.cache) {
 			if (/fixture-fail/.test(key.toString())) {
 				throw new Error('require cache still contained: ' + key);
 			}
 		}
-
-		cb();
 		done();
-	}));
-
-	stream.on('error', function () {});
+	});
 	stream.write(new gutil.File({path: 'fixture-fail.js'}));
 	stream.end();
 });
@@ -93,20 +83,14 @@ it('should clear cache after failing run', function (done) {
 it('should clear cache after mocha threw', function (done) {
 	var stream = mocha();
 
-	stream.pipe(through.obj(function (file, enc, cb) {
-		cb();
-	}, function (cb) {
+	stream.on('error', function () {
 		for (var key in require.cache) {
 			if (/fixture-pass/.test(key.toString()) || /fixture-throws/.test(key.toString())) {
 				throw new Error('require cache still contained: ' + key);
 			}
 		}
-
-		cb();
 		done();
-	}));
-
-	stream.on('error', function () {});
+	});
 	stream.write(new gutil.File({path: 'fixture-pass.js'}));
 	stream.write(new gutil.File({path: 'fixture-throws.js'}));
 	stream.end();
@@ -115,20 +99,14 @@ it('should clear cache after mocha threw', function (done) {
 it('should clear cache after mocha threw uncaught exception', function (done) {
 	var stream = mocha();
 
-	stream.pipe(through.obj(function (file, enc, cb) {
-		cb();
-	}, function (cb) {
-		for (var key in require.cache) {
-			if (/fixture-pass/.test(key.toString()) || /fixture-throws/.test(key.toString())) {
-				throw new Error('require cache still contained: ' + key);
+	stream.on('error', function () {
+			for (var key in require.cache) {
+				if (/fixture-pass/.test(key.toString()) || /fixture-throws/.test(key.toString())) {
+					throw new Error('require cache still contained: ' + key);
+				}
 			}
-		}
-
-		cb();
-		done();
-	}));
-
-	stream.on('error', function () {});
+			done();
+	});
 	stream.write(new gutil.File({path: 'fixture-pass.js'}));
 	stream.write(new gutil.File({path: 'fixture-throws-uncaught.js'}));
 	stream.end();

